examples/simple_demo_qml/ThreadRenderer.cpp

```diff
@@ -321,32 +321,84 @@ void TextureNode::NewTexture(int _id, const QSize &_size)
 void TextureNode::PrepareNode()
 {
     this->mutex.lock();
+    // new texture ID from render engine (Ogre)
     int newId = this->id;
-    QSize size = this->size;
+    QSize newSize = this->size;
     this->id = 0;
     this->mutex.unlock();
+
     if (newId)
     {
         delete this->texture;
         this->texture = nullptr;
         // note: include QQuickWindow::TextureHasAlphaChannel if the rendered content
         // has alpha.
+
+        GLuint texForQt = newId;
+
+        if (this->graphicsAPI == gz::rendering::GraphicsAPI::OPENGL)
+        {
+            // intermediate FBO for copying texture data
+            GLuint rhiFbo = 0;
+            // internal texture ID to hold texture data from render engine texture,
+            // in linear GL_RGB format
+            GLuint rhiId = 0;
```

Comment:
can you make these ids member variables so that we can delete the fbo and texture in the destructor when the app exits? e.g. see code [here](https://github.com/gazebosim/gz-gui/blob/2496041b72e6cc29b2b520ef5aca0c622618cb0a/src/plugins/minimal_scene/MinimalSceneRhiOpenGL.cc#L209-L214)